src/Visitor/Translation/TranslationCheckVisitor.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+
+namespace Torr\PrismicApi\Visitor\Translation;
+
+use Torr\PrismicApi\Data\Value\DocumentLinkValue;
+use Torr\PrismicApi\Structure\Field\LinkField;
+use Torr\PrismicApi\Structure\Field\RichTextField;
+use Torr\PrismicApi\Structure\PrismicTypeInterface;
+use Torr\PrismicApi\Visitor\DataVisitorInterface;
+
+class TranslationCheckVisitor implements DataVisitorInterface
+{
+	/** @var TranslationCheckIssue[] */
+	private array $issue = [];
+
+	public function __construct(
+		private readonly string $documentLocale,
+	) {}
+
+
+	/**
+	 */
+	public function onDataVisit(
+		PrismicTypeInterface $field,
+		mixed $data,
+	) : void
+	{
+
+		if ($field instanceof RichTextField)
+		{
+			foreach ($data as $item)
+			{
+				if (!empty($item))
+				{
+					foreach ($item["spans"] as $span) {
+						if ("hyperlink" === $span["type"])
+						{
+							if($span["data"]["url"] instanceof DocumentLinkValue && $span["data"]["url"]->getTargetLocale() !== $this->documentLocale)
+							{
+								$this->issue[] = [
+									"severity" => "ERROR",
+									"message" => "RTE - Wrong locale in Link.",
+									"field" => $field,
+								];
+							}
+						}
+
+						if (!empty($span["type"]))
```

Comment:
We should only add it once per RTE field. Right now we would add it per paragraph